services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/UserGroupController.java

```diff
@@ -0,0 +1,265 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Slf4j
+@RequestMapping("/groups")
+@RequiredArgsConstructor
+public class UserGroupController {
+
+  private final GroupRepository groupRepository;
+
+  private final UserGroupsRepository userGroupsRepo;
+
+  @Value("${one.owner.msg}")
+  private String oneOwnerMsg;
+
+  @Value("${user.group.not.found}")
+  private String usrGrpNotFound;
+
+  @Value("${only.grp.owner.access}")
+  private String grpOwnerAccess;
+
+  @PostMapping("{id}/user-groups")
+  public ResponseEntity<Object> createRole(
+    @Valid @RequestBody UserGroup userGroup,
+    @PathVariable("id") UUID id
+  ) {
+    Optional<UserGroup> usrGr;
+    Group svdGroup;
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      svdGroup = group.get();
+      if (userGroup.getGroup().getId() == null) {
+        userGroup.getGroup().setId(svdGroup.getId());
+      }
+      usrGr =
+        userGroupsRepo.findByGroupIdAndUserTenantId(
+          userGroup.getGroup().getId(),
+          userGroup.getUserTenant().getId()
+        );
+      if (!usrGr.isPresent()) {
+        usrGr = (Optional.ofNullable(userGroupsRepo.save(userGroup)));
+        group.get().setModifiedOn(usrGr.get().getModifiedOn());
+        groupRepository.save(group.get());
+      }
+    } else {
+      throw new ResponseStatusException(
+        HttpStatus.NOT_FOUND,
+        CommonConstants.NO_GRP_PRESENT + id
+      );
+    }
+    return new ResponseEntity<>(usrGr.get(), HttpStatus.CREATED);
+  }
+
+  @Transactional
+  @PatchMapping("{id}/user-groups/{userGroupId}")
+  public ResponseEntity<Object> updateAll(
+    @PathVariable("id") UUID id,
+    @RequestBody UserGroup userGroup,
+    @PathVariable("userGroupId") UUID userGroupId
+  ) {
+    /** fetch value in Group against primary key and also to
+         and to update modifiedOn parameter*/
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      Count count = Count
+        .builder()
+        .totalCount(userGroupsRepo.countByGroupIdAndIsOwner(id, true))
+        .build();
+      if (count.getTotalCount() == 1) {
+        Optional<UserGroup> usrGrp =
+          userGroupsRepo.findByGroupIdAndIdAndIsOwner(id, userGroupId, true);
+
+        if (usrGrp.isPresent() && !usrGrp.get().isOwner()) {
+          throw new ResponseStatusException(HttpStatus.FORBIDDEN, oneOwnerMsg);
+        }
+      }
+      if (userGroup.getId() == null) {
```

Comment:
what if userGroup id is provided by user but its not exists in db it will create a new record instead of updating, right? i think you should remove the condition and replace id for both cases of id is null or not null or directly set null